internal/repository/article/article.go

```diff
@@ -0,0 +1,85 @@
+package article
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"news/internal/entity"
+	"news/internal/repository/mgdb"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+type ArticleRep struct {
+	articleTitle *mongo.Collection
+	article      *mongo.Collection
+}
+
+func NewArticleRep(client *mgdb.MongoDriver) *ArticleRep {
+	return &ArticleRep{
+		articleTitle: client.CreateCollection("news", "articleTitle"),
+		article:      client.CreateCollection("news", "article"),
+	}
+}
+
+func (a *ArticleRep) SaveArticleList(ctx context.Context, articleList *entity.ArticleList) ([]int64, error) {
+	articles := make([]interface{}, 0, len(articleList.ArticleList.Articles))
```

Comment:
usually you're supposed to have different models for transport and repository layers plus converters from one to another.

https://threedots.tech/post/common-anti-patterns-in-go-web-applications/ - search by "Anti-pattern: The Single Model"

Reply:
oh i get it
fixed